src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -569,6 +578,7 @@ MulticopterPositionControl::parameters_update(bool force)
 		/* takeoff and land velocities should not exceed maximum */
 		_params.tko_speed = fminf(_params.tko_speed, _params.vel_max(2));
 		_params.land_speed = fminf(_params.land_speed, _params.vel_max(2));
+		_params.land_final_approach_speed = fminf(_params.land_final_approach_speed, _params.vel_max(2));
```

Comment:
Could you also constrain this to be lower than land_speed? Otherwise your deceleration will do the opposite if set wrong.